### ISSUE TITLE
Installing Flux more easily

### ISSUE THREAD

**Issue by vsoch** [flux-framework/flux-operator#118]
We really need to be able to remove the requirement that flux is installed in the container, either possibly via an app image, layers injected in at runtime, or similar. Do you have any suggestions @ChristianKniep or @pditommaso? Flux has (of course) binaries, config files, and uses system libraries, so it's not so trivial as building one binary and klunking it into a layer and then bin!

**Comment by pditommaso** [comment]
This is what Wave is made for. You can make an [API request](https://wavecontainers.io/api/) and get the for a Wave container image name that represents your source image, plus the layer including Flux. 

The only limitation I see is that Wave container image so far are ephemeral, in the meaning it expires after a few hours. 

However thinking to your use case, it could be useful, and simple, to add the possibility to have a permanent container name.

**Comment by ChristianKniep** [comment]
Yup, MetaHub will solve the same issue. I reckon that it's going to be hard otherwise.

On one hand you want to have images that are agnostic to the execution environment and the orchestrator as they are most likely upstream images out of your control. The upstream community does not have an incentive to include all the tools needed downstream (because it would be a stupid idea).

**Comment by vsoch** [comment]
So MetaHub and Wave are fairly similar it seems? They both take a pull request and modify the manifest and return a different one. It sounds like I should implement something (simple) like this and serve it alongside the MiniCluster? :thinking:

**Comment by pditommaso** [comment]
<img src="https://media0.giphy.com/media/3cXmze4Y8igXdnkc3U/giphy.gif"/>

**Comment by vsoch** [comment]
Why? Because I need the functionality to add flux to containers at runtime, and I can’t rely on private or paid for services?

**Comment by vsoch** [comment]
Oh I could use Wave? I didn't realize that! I thought it was just integrated into Nextflow / Tower?

**Comment by vsoch** [comment]
okay sweet! I'll try that later today. Thank you @pditommaso and @ChristianKniep !

![thanks-a-bunch](https://user-images.githubusercontent.com/814322/225644682-f60f4317-20b3-4e5d-9a2c-38e13c7a8ec3.gif)
